src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+            Inherits Analyzer(Of QueryExpressionSyntax, ExpressionSyntax)
+
+            Public Sub New(semanticModel As SemanticModel, cancellationToken As CancellationToken)
+                MyBase.New(semanticModel, cancellationToken)
+            End Sub
+
+            Protected Overrides ReadOnly Property Title() As String
+                Get
+                    Return VBFeaturesResources.Convert_linq_query_to_linq_method
+                End Get
+            End Property
+
+            Protected Overrides Function TryConvert(source As QueryExpressionSyntax) As ExpressionSyntax
+                Dim expression As ExpressionSyntax
+                expression = Nothing
+
+                Dim symbolInfo = _semanticModel.GetSymbolInfo(source, _cancellationToken)
+                For Each clause In source.Clauses
+                    expression = ProcessQueryClause(expression, clause)
+                    If expression Is Nothing Then
+                        Return Nothing
+                    End If
+                Next
+
+                Return expression.WithTrailingTrivia(source.GetTrailingTrivia())
+            End Function
+
+            Private Function ProcessQueryClause(expression As ExpressionSyntax, queryClause As QueryClauseSyntax) As ExpressionSyntax
+                Select Case (queryClause.Kind())
+                    Case SyntaxKind.FromClause
+                        Return ProcessFromClause(expression, DirectCast(queryClause, FromClauseSyntax))
+                    Case SyntaxKind.DistinctClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Distinct))
+                    Case SyntaxKind.WhereClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Where), DirectCast(queryClause, WhereClauseSyntax).Condition)
+                    Case SyntaxKind.SkipClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Skip), DirectCast(queryClause, PartitionClauseSyntax).Count)
+                    Case SyntaxKind.TakeClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Take), DirectCast(queryClause, PartitionClauseSyntax).Count)
+                    Case SyntaxKind.OrderByClause
+                        Return ProcessOrderByClause(expression, DirectCast(queryClause, OrderByClauseSyntax))
+                    Case SyntaxKind.SelectClause
+                        Return ProcessSelectClause(expression, DirectCast(queryClause, SelectClauseSyntax))
+                    Case SyntaxKind.GroupJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.SimpleJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.LetClause
+                        Return Nothing ' Too complicated for query method syntax.
```

Comment:
>Return Nothing ' Too complicated for query method syntax. [](start = 24, length = 57)

I am not sure why do we think Let is too complicated

Reply:
probably not too complicated but suggest to consider is aside if not trivial. replaced with a TODO to [https://github.com/dotnet/roslyn/issues/25112](https://github.com/dotnet/roslyn/issues/25112)

---
In reply to: [171636458](https://github.com/dotnet/roslyn/pull/24759#discussion_r171636458) [](ancestors = 171636458)